modules/apps/asset/asset-publisher-web/src/main/java/com/liferay/asset/publisher/web/internal/portlet/AssetPublisherPortlet.java

```diff
@@ -490,9 +510,170 @@ protected boolean isSessionErrorException(Throwable throwable) {
 	@Reference
 	protected RequestContextMapper requestContextMapper;
 
+	@Reference
+	protected RSSExporter rssExporter;
+
 	@Reference
 	protected SegmentsEntryRetriever segmentsEntryRetriever;
 
+	@Reference
+	protected SyndModelFactory syndModelFactory;
+
+	private String _exportToRSS(
+			PortletRequest portletRequest, PortletResponse portletResponse,
+			String name, String format, double version, String displayStyle,
+			String linkBehavior, List<AssetEntry> assetEntries)
+		throws Exception {
+
+		ThemeDisplay themeDisplay = (ThemeDisplay)portletRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+
+		SyndFeed syndFeed = syndModelFactory.createSyndFeed();
+
+		syndFeed.setDescription(name);
+
+		List<SyndEntry> syndEntries = new ArrayList<>();
+
+		syndFeed.setEntries(syndEntries);
+
+		for (AssetEntry assetEntry : assetEntries) {
+			SyndEntry syndEntry = syndModelFactory.createSyndEntry();
+
+			syndEntry.setAuthor(portal.getUserName(assetEntry));
+
+			SyndContent syndContent = syndModelFactory.createSyndContent();
+
+			syndContent.setType(RSSUtil.ENTRY_TYPE_DEFAULT);
+
+			String value = null;
+
+			String languageId = language.getLanguageId(portletRequest);
+
+			if (displayStyle.equals(RSSUtil.DISPLAY_STYLE_TITLE)) {
+				value = StringPool.BLANK;
+			}
+			else {
+				value = assetEntry.getSummary(languageId, true);
+			}
+
+			syndContent.setValue(value);
+
+			syndEntry.setDescription(syndContent);
+
+			String link = _getEntryURL(
+				portletRequest, portletResponse, linkBehavior, assetEntry);
+
+			syndEntry.setLink(link);
+
+			syndEntry.setPublishedDate(assetEntry.getPublishDate());
+			syndEntry.setTitle(assetEntry.getTitle(languageId, true));
+			syndEntry.setUpdatedDate(assetEntry.getModifiedDate());
+			syndEntry.setUri(link);
+
+			syndEntries.add(syndEntry);
+		}
+
+		syndFeed.setFeedType(RSSUtil.getFeedType(format, version));
```

Comment:
Hey @MrJohn1911, I know it was not included by you but try to remove the not needed declared variables.